x-pack/plugins/security/server/lib/authorization/privilege_serializer.ts

```diff
@@ -6,73 +6,73 @@
 
 const featurePrefix = 'feature_';
 const spacePrefix = 'space_';
-const reservedPrivilegeNames = ['all', 'read'];
+const minimumPrivilegeNames = ['all', 'read'];
+const spaceMinimumPrivilegeNames = minimumPrivilegeNames.map(
+  privilegeName => `${spacePrefix}${privilegeName}`
+);
+const deserializeFeaturePrivilegeRegexp = new RegExp(
+  `^${featurePrefix}([a-zA-Z0-9_-]+)\\.([a-zA-Z0-9_-]+)$`
+);
+
+interface FeaturePrivilege {
+  featureId: string;
+  privilege: string;
+}
 
 export class PrivilegeSerializer {
-  public static serializeGlobalReservedPrivilege(privilegeName: string) {
-    if (!reservedPrivilegeNames.includes(privilegeName)) {
-      throw new Error('Unrecognized global reserved privilege');
+  public static isGlobalMinimumPrivilege(privilegeName: string) {
+    return minimumPrivilegeNames.includes(privilegeName);
+  }
+
+  public static isSpaceMinimumPrivilege(privilegeName: string) {
+    return spaceMinimumPrivilegeNames.includes(privilegeName);
+  }
+
+  public static serializeGlobalMinimumPrivilege(privilegeName: string) {
+    if (!minimumPrivilegeNames.includes(privilegeName)) {
+      throw new Error('Unrecognized global minimum privilege');
     }
 
     return privilegeName;
   }
 
-  public static serializeSpaceReservedPrivilege(privilegeName: string) {
-    if (!reservedPrivilegeNames.includes(privilegeName)) {
-      throw new Error('Unrecognized space reserved privilege');
+  public static serializeSpaceMinimumPrivilege(privilegeName: string) {
+    if (!minimumPrivilegeNames.includes(privilegeName)) {
```

Comment:
Should this read `if (!spaceMinimumPrivilegeNames.includes(privilegeName))...`?

In any case, same nit here: what do you think about using `isSpaceMinimumPrivilege` here, instead of duplicating the admittedly small amount of logic?

Reply:
Unserialized, space and global privileges are both `[ 'all', 'read' ]`, so we don't want to be using `isSpaceMinimumPrivilege`, but using `minimumPrivilegeNames` is confusing... lemme try and clean up the naming of these variables.